runtime/runtime/src/prefetch.rs

```diff
@@ -195,22 +220,96 @@ impl TriePrefetcher {
         arg: &[u8],
     ) -> Result<(), PrefetchError> {
         if let Ok(json) = serde_json::de::from_slice::<serde_json::Value>(arg) {
-            if json.is_object() {
-                if let Some(list) = json.get("steps_batch") {
-                    if let Some(list) = list.as_array() {
-                        for tuple in list.iter() {
-                            if let Some(tuple) = tuple.as_array() {
-                                if let Some(user_account) = tuple.first().and_then(|a| a.as_str()) {
-                                    let hashed_account =
-                                        sha2::Sha256::digest(user_account.as_bytes()).into_iter();
-                                    let mut key = vec![0x74, 0x00];
-                                    key.extend(hashed_account);
-                                    let trie_key = TrieKey::ContractData {
-                                        account_id: account_id.clone(),
-                                        key: key.to_vec(),
-                                    };
-                                    near_o11y::io_trace!(count: "prefetch");
-                                    self.prefetch_trie_key(trie_key)?;
+            if let Some(list) = get_inner_json(&json, "steps_batch") {
```

Comment:
nit: ditto on using the `let ... else` pattern for early returns to flatten the code

Reply:
same in other methods below